src/vectorstore/surrealdb_store.py

```
@@ -414,6 +429,105 @@ def _sanitize_metadata(self, metadata: Dict[str, Any]) -> Dict[str, Any]:
+            raise VectorStoreException("Client not connected")
+
+        try:
+            # Search for similar questions using cosine similarity
+            query = f"""
+                SELECT id, question, answer, metadata,
+                       vector::similarity::cosine(question_embedding, {query_embedding}) AS score
+                FROM qa_history
+                WHERE vector::similarity::cosine(question_embedding, {query_embedding}) >= {threshold}
+                ORDER BY score DESC
+                LIMIT 1;
+            """
```

Ruff (0.15.5)

[error] Possible SQL injection vector through string-based query construction

(S608)